### DISCUSSION TITLE
Surface setup for racing

### DISCUSSION THREAD

**Discussion by Divefire** [EdgeTX/edgetx#5709] [Q&A]
I'd like to use my MT12 to replicate the functionality of a racing surface radio. My first thought is to start in the 2005 era transmitters and see what of the functionality and display can be replicated with EdgeTX as of now. The reason for a earlier generation of focus is to limit the scope to start with and give a realistic goal to see what is possible. 

My starting models for this are the Futaba 3PK and Sanwa M11 as both seem to have useful features and display layout, plus were popular radios of the time. 

The feature list of the radios is as follows: 

- [ ] Dual Rate Steering; 2nd Dual Rate; Separate values for left and right. 
- [ ] End Point Adjustment; Separate values for steering left, right and throttle forward, back
- [ ] Exponential; with real time graphs (this is not shown in the mixer screen currently on EdgeTX
- [ ] Speed; values that only effect one direction and not reverse it. 

My current understanding of EdgeTX is that there are issues trying to get these working the way described but I would like to be proven otherwise. 

- [ ] Adjustable Rate Control (steering and throttle)
- [ ] Anti-Lock Braking (ABS)
- [ ] Traction Control

These I believe can be made via various combinations of mixers, GF,  and LS. However when it comes to displaying that information via a LUA script I'm unsure how feasible that is. 

Some or all of these functions can be set to be adjusted via buttons, toggles, sliders etc, with out going into the menu functions which is a real boon for making adjustments when racing. These functions are: 

- [x] Dual rate
- [x] 2nd dual rate
- [x] Exponential 
- [ ] Speed
- [x] ABS 
- [x] Traction Control
- [x] Trim
- [x] Sub-trim
- [x] Lap start/reset

Speed I believe is currently not possible because Delay and Slow are not adjustable by GFs. 

So my broad question is can anyone point me in the directions of how to set an MT12 up like this and what LUA magic I need to perform to get a nice display for it?

**Comment by inventor7777** [comment]
Do you mean that you want to get those mentioned features _working_, or you want the actual _information_ for those said functions to be on the main display?

**Reply by Divefire** [reply]
I should have added, those items with a tick are settable in EdgeTX now, those with out are not. So I would like all of them to be able to be created by the user and I would also like that information able to be displayed on the main screen when adjusting the value. 

The latter is not always the case in the various screen of EdgeTX (setting the expo curve in mixers for instance) and I'm not versed in lua enough to know if lua can be used to display that information on the main screen, let alone on demand.
